### PR TITLE
Skip alert for RelVals with deletable pileup

### LINKED ISSUE #11279: Do not create notifications in MSRuleCleaner for RelVal pileup
**Impact of the new feature**
MSRuleCleaner

**Is your feature request related to a problem? Please describe.**
Follow up from this GH issue:
https://github.com/dmwm/WMCore/issues/11216

it turns out Release Validation workflows create more alerts than I would expect. We have been getting around 10 email notifications every day. So I suggest to simply disable such alerts for `RelVal` workflows.

**Describe the solution you'd like**
MSRuleCleaner makes a prometheus notification whenever it finds a pileup that is eligible for deletion (which have no workflows locking that pileup anymore). 

My suggestion is to make these notifications only for non-RelVal workflows. Whenever a RelVal workflow is found with a pileup eligible for rule deletion, we should simply make a record in the log and move on.

**Describe alternatives you've considered**
Keep getting all (10/day) these alerts. 

**Additional context**
None

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #11279 

#### Status
not-tested

#### Description
Change was supposed to be very basic as of not creating alert notifications for Release Validation workflows. However, we cannot distinguish RelVals from non-RelVals in MSRuleCleaner, hence the code refactoring avoiding to duplicate code.

Note that this PR adds a new `SubRequestType` key/value to the MSRuleCleanerWflow data structure.

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
None

#### External dependencies / deployment changes
None

### CODE REVIEW

**Comment by amaltaro** [comment]
test this please

**Review by todor-ivanov — APPROVED** [review]
looks good. Thanks @amaltaro

File: src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleanerWflow.py [hunk 1]
```diff
@@ -200,6 +201,7 @@ def docSchema(self):
         docTemplate = [
             ('RequestName', None, (bytes, str)),
             ('RequestType', None, (bytes, str)),
+            ('SubRequestType', None, (bytes, str)),
```
Comment (vkuznet) [inline src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleanerWflow.py:204]: Due to this change I want to raise your awareness of consequences of having multiple data-type assignments to a single variable. Why do you do that? Such assignment only creates other problems down in the road making checks, and code to do extra work and hide issue with data-types. All conversions should be done in place and attributes in a document should have single data type. But it is my view and I doubt you'll make the changes. But as reviewer I want to want you about it.

**Review by vkuznet — APPROVED** [review]

File: src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleanerWflow.py (hunk 1 above)
Comment (amaltaro) [reply src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleanerWflow.py:204]: Byte and unicode strings was very hard to deal while making WMCore compatible with py2 and py3. As we start removing python2 code and making WMCore python3-only, we should definitely revisit strings and try to converge to a single type as much as we can in the project. Good point!

For now, I am just following the stream here.

**Pull request merged**